### ISSUE TITLE
Support Jackson JsonAnyGetter

### ISSUE THREAD

**Issue by DavidLeoni** [immutables/immutables#185]
Seems like it's not possible to store unknown fields in immutable objects when deserializing with jackson. The following test serializes correctly to {"x":"y"}, but then on deserializing throws this exception:

`com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "x" (class trials.ImmutableObj$Json), not marked as ignorable (one known property: "other"])`

```

@Value.Immutable
@JsonSerialize(as = ImmutableObj.class)
@JsonDeserialize(as = ImmutableObj.class)
abstract class Obj {    
    
    @JsonAnyGetter
    @Value.Parameter
    public abstract Map<String, String> other();
    
}


public class MyTest {
    
    
    @Test
    public void test1() throws Exception{
	ObjectMapper om = new ObjectMapper();	
	
	HashMap<String, Object> map = new HashMap();
	map.put("x",  "y");
	ImmutableObj imm = ImmutableObj.builder().putOther("x", "y").build();
	
	String json = om.writeValueAsString(imm);
	
	System.out.println(json);
	
	ImmutableObj ap = om.readValue(json, ImmutableObj.class);
    }
    
}
```

Maybe the generated ImmutableObj.Json class should contain a `@JsonAnySetter` method.

**Comment by elucash** [comment]
It should contain it. Need to add support for this. Thank you for the report!

**Comment by elucash** [comment]
This should be supported now, released in 2.1.1. Thank you!
